src/shared-validator.cc

```diff
@@ -76,6 +76,21 @@ Result SharedValidator::OnArrayType(const Location&, TypeMut field) {
   return Result::Ok;
 }
 
+Result SharedValidator::EndTypeSection() {
+  Result result = Result::Ok;
```

Comment:
This is a new event in SharedValidator. This check could be done in `OnFuncType`, but then we needed the `OnTypeCount` event, which is also not used by the validator.

Since the Google v8 engine has separate reference types for array/struct, we might need to check the reference type in the future as well, and we cannot do that before reading all types.